lib/ddtrace/profiling/recorder.rb

```diff
@@ -19,11 +19,16 @@ def push(event)
         @buffers[event.class].push(event)
       end
 
-      def pop(event_class)
-        raise UnknownEventError, event_class unless @buffers.key?(event_class)
-        @buffers[event_class].pop
+      def pop
```

Comment:
Per Python implementation, it looks like we need to pop all events from all buffers, for use in the `Scheduler`, hence this change.

Reply:
Is that ok on thread-safety?

The typical use case is that the collector will continue to push events while this is pop.

(Just asking, I'm not Ruby-competent enough to assess).

Reply:
Should be safe: the `pop`/`push` in the buffers themselves are mutexed, and should be blocking operations. If another thread is either reading or writing to one of the buffers, the `Scheduler` will block until that read/write is complete.